commands/init.js

```diff
@@ -88,47 +132,180 @@ function copyFiles (srcPath, destPath) {
-
-    if (typescript) {
-      // Remove non-typescript source files from template
+    if (typeScript) {
       fs.unlinkSync(`${currentDirectory}/${folderName}/src/main.js`);
       fs.unlinkSync(`${currentDirectory}/${folderName}/src/app.js`);
       // Copy typescript template overlay over existing template files
       const pathSuffix = (appType === 'Components') ? '_components' : '';
       templatePath = `${__dirname}/../template_overlay_typescript${pathSuffix}`;
       copyFiles(templatePath, `${currentDirectory}/${folderName}`);
     }
-  });
+  }).catch((err) => console.log(red, err, normal))
+    // Add this callback for testing purpose - inquirer doesn't provide functionality to reset values
+    // after every test and caches them
+    .finally(() => {
+      folderName = null;
+      visibleName = null;
+      appType = null;
+      packageName = null;
+      target = null;
+      typeScript = null;
+    });
 };
+
+function preparePlatforms (destPath) {
+  let android = target.includes('ANDROID');
+  let iOS = target.includes('IOS');
+  let lumin = target.includes('LUMIN');
+  let isReact = (target.includes('IOS') || target.includes('ANDROID'));
+  if (!iOS) {
+    if (fs.existsSync(`${destPath}/reactnative/ios`)) {
+      util.removeFilesRecursively(`${destPath}/reactnative/ios`);
+    }
+  }
+  if (!android) {
+    if (fs.existsSync(`${destPath}/reactnative/android`)) {
+      util.removeFilesRecursively(`${destPath}/reactnative/android`);
+    }
+  } else {
+    util.createAndroidLocalProperties(destPath);
+  }
+  if (!isReact) {
+    if (fs.existsSync(`${destPath}/reactnative`)) {
+      util.removeFilesRecursively(`${destPath}/reactnative`);
+    }
+  }
+  if (!lumin) {
+    if (fs.existsSync(`${destPath}/lumin`)) {
+      util.removeFilesRecursively(`${destPath}/lumin`);
+    }
+  }
+}
+
+function isComponentsAndAtLeastOneTarget (appType, argTarget) {
+  return (appType === 'Components' && argTarget && argTarget.some(substring => {
+    return targetPlatforms.includes(substring);
+  }));
+}
+
+function setFolderName (name) {
+  if (name && util.isValidFolderName(name)) {
+    folderName = name;
+    if (!visibleName) {
+      visibleName = folderName;
+    }
+  }
+}
+
+function setVisibleName (name) {
+  if (name) {
+    visibleName = name;
+  }
+}
+
+function setAppType (type) {
+  if (util.isValidAppType(type)) {
+    appType = type;
+  }
+}
+
+function setPackageName (name) {
+  if (name && util.isValidPackageId(name)) {
+    packageName = name;
+  }
+}
+
+function isLandscapeOrImmersive (folderName, packageName, appType) {
+  return packageName && folderName && appType && (appType === 'Immersive' || appType === 'Landscape');
+}
+
+function isComponents (folderName, packageName, appType) {
+  return folderName && packageName && appType && appType === 'Components';
+}
+
+function setTarget (appType, argTarget) {
+  if (argTarget && Array.isArray(argTarget)) {
+    target = argTarget.map(toUpperCase);
```

Comment:
this can be re-written as
```
target = argTarget.map((string)=>{
  return string.toUpperCase();
});
```
and now you don't need the toUpperCase function defined, which is only used here.